docs/sorting-algorithms/selection-sort.mdx

```diff
@@ -0,0 +1,206 @@
+---
+title: Selection Sort
+slug: /sorting-algorithms/selection-sort
+---
+
+# Introduction
+
+Selection Sort is a sorting algorithm which finds out the smallest element in unsorted part in each iteration and put it to the last of sorted part.
+It does less memory writes as compared to other sorting algorithms.
+
+# Algorithm
+
+In Selection Sort, we follow the following algorithm:
+
+* First we will find the smallest element in the range [0,n-1] and swap it with the element at 0th index
+
+* Then we will find the smallest element int the range [1,n-1] and swap it with the element at 1st index
+
+* We keep on increasing the left part of the range until it reaches n-1 and do the above mentioned process
+
+# Explanation for Selection Sort
+
+### Example 1
+
+```
+   0   1   2   3
+ | 4 | 1 | 2 | 7 |
+
+ [0,3] min_idx=1 
+ swap(arr[min_idx],arr[0]);
+
+ | 1 | 4 | 2 | 7 |
+
+ [1,3] min_idx=2
+ swap(arr[min_idx],arr[1]);
+
+ | 1 | 2 | 4 | 7 |
+
+ [2,3] min_idx=2 yar 
+ swap(arr[min_idx],arr[2]);
+```
+### Example 2
+
+```
+   0   1   2   3
+ | 1 | 8 | 5 | 2 |
+
+ [0,3] min_idx=0
+ swap(arr[min_idx],arr[0]);
+
+ | 1 | 8 | 5 | 2 |
+
+ [1,3] min_idx=3
+ swap(arr[min_idx],arr[1]);
+
+ | 1 | 2 | 4 | 8 |
+
+ [2,3] min_idx=2 
+ swap(arr[min_idx],arr[2]);
+```
+
+# Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+    { label: 'Java', value: 'java', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+#include<iostream>
+using namespace std;
+void swapping(int &a, int &b) {         //swap the content of a and b
+   int temp;
+   temp = a;
+   a = b;
+   b = temp;
+}
+
+
+void display(int *array, int size) {
+   for(int i = 0; i<size; i++)
+      cout << array[i] << " ";
+   cout << endl;
+}
+
+
+void selection_sort(int *array, int size) {
+   int i, j, imin;
+   for(i = 0; i<size-1; i++) {
+      imin = i;   //get index of minimum data
+      for(j = i+1; j<size; j++)
+         if(array[j] < array[imin])
+            imin = j;
+         //placing in correct position
+         swap(array[i], array[imin]);
+   }
+}
+int main() {
+   int n;
+   cout << "Enter the number of elements: ";
+   cin >> n;
+   int arr[n];           //create an array with given number of elements
+   cout << "Enter elements:" << endl;
+   for(int i = 0; i<n; i++) {
+      cin >> arr[i];
+   }
+   cout << "Array before Sorting: ";
+   display(arr, n);
+   selection_sort(arr, n);
+   cout << "Array after Sorting: ";
+   display(arr, n);
+}
+```
```

Comment:
This piece of code seems to be copied from [here](https://github.com/bhavyagupta29/basic/blob/main/selection%20sort). I guess the plagiarism existed in NeoAlgo itself. Can you write your own implementation?

Reply:
![p](https://user-images.githubusercontent.com/56471793/114140237-a2de7c80-992d-11eb-94a6-7eb13dd09ef0.PNG)
Here it is written that we can use code from neoalgo

Reply:
Yeah. I understand this. Before our review process was laxly due to which a lot of plagiarized and bad code. You can make the changes and then make a PR back to NeoAlgo so that you can be equally compensated.

Reply:
done